src/main/java/org/nuxeo/shibboleth/invitation/ShibbolethUserMapper.java

```diff
@@ -99,7 +99,8 @@ public NuxeoPrincipal getOrCreateAndUpdateNuxeoPrincipal(Object userObject, bool
             userDoc = findUser(userManager.getUserIdField(), email);
         }
         if (userDoc != null && userName != null) {
-            updateACP(userName, email, userDoc);
+            DocumentModel finalUserDoc = userDoc; // Effectively final
+            TransactionHelper.runInTransaction(() -> updateACP(userName, email, finalUserDoc));
```

Comment:
Is this needed given that NXP-22463 is only for 9.3?

Reply:
Right, I forgot to remove this from the cherry-pick.